lms/djangoapps/teams/views.py

```diff
@@ -0,0 +1,508 @@
+    """
+        **Use Cases**
+
+            Get or create a course team.
+
+        **Example Requests**:
+
+            GET /api/team/v0/teams
+
+            POST /api/team/v0/teams
+
+        **Query Parameters for GET**
+
+            * course_id: Filters the result to teams belonging to the given
+              course. Required.
+
+            * topic_id: Filters the result to teams associated with the given
+              topic.
+
+            * text_search: Currently not supported.
+
+            * order_by: Must be one of the following:
+
+                * name: Orders results by case insensitive team name (default).
+
+                * open_slots: Orders results by most open slots.
+
+                * last_activity: Currently not supported.
+
+            * page_size: Number of results to return per page.
+
+            * page: Page number to retrieve.
+
+            * include_inactive: If true, inactive teams will be returned. The
+              default is to not include inactive teams.
+
+        **Response Values for GET**
+
+            If the user is logged in and enrolled, the response contains:
+
+            * count: The total number of teams matching the request.
+
+            * next: The URL to the next page of results, or null if this is the
+              last page.
+
+            * previous: The URL to the previous page of results, or null if this
+              is the first page.
+
+            * num_pages: The total number of pages in the result.
+
+            * results: A list of the teams matching the request.
+
+                * id: The team's unique identifier.
+
+                * name: The name of the team.
+
+                * is_active: True if the team is currently active. If false, the
+                  team is considered "soft deleted" and will not be included by
+                  default in results.
+
+                * course_id: The identifier for the course this team belongs to.
+
+                * topic_id: Optionally specifies which topic the team is associated
+                  with.
+
+                * date_created: Date and time when the team was created.
+
+                * description: A description of the team.
+
+                * country: Optionally specifies which country the team is
+                  associated with.
+
+                * language: Optionally specifies which language the team is
+                  associated with.
+
+                * membership: A list of the users that are members of the team.
+                  See membership endpoint for more detail.
+
+            For all text fields, clients rendering the values should take care
+            to HTML escape them to avoid script injections, as the data is
+            stored exactly as specified. The intention is that plain text is
+            supported, not HTML.
+
+            If the user is not logged in and enrolled in the course specified by
+            course_id or is not staff, a 403 error is returned.
```

Comment:
Will need to clarify what "not staff" means-- we are talking about global staff here, not course staff.